.github/workflows/ci.yml

```diff
@@ -87,6 +87,56 @@ jobs:
         if: steps.cache.outputs.cache-hit != 'true' && matrix.cache == 'true'
         run: rm -rf ~/.m2/repository/io/trino/trino-*
 
+  reproducible-build:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # checkout all commits, as the build result depends on `git describe` equivalent
+          ref: |
+            ${{ github.event_name == 'repository_dispatch' &&
+                github.event.client_payload.pull_request.head.sha == github.event.client_payload.slash_command.args.named.sha &&
+                format('refs/pull/{0}/head', github.event.client_payload.pull_request.number) || '' }}
+      - uses: ./.github/actions/setup
+        with:
+          cache: 'false'
+          cleanup-node: true
+      - name: First build to a local repository
+        run: |
+          export MAVEN_OPTS="${MAVEN_INSTALL_OPTS}"
+          $MAVEN clean install ${MAVEN_FAST_INSTALL} -Dmaven.repo.local=build_1
+      - name: Second build to a local repository
+        run: |
+          export MAVEN_OPTS="${MAVEN_INSTALL_OPTS}"
+          $MAVEN clean install ${MAVEN_FAST_INSTALL} -Dmaven.repo.local=build_2
+      - name: Compare builds
+        run: |
+          diff \
+              --exclude maven-metadata-local.xml \
+              --exclude=_remote.repositories \
+              --exclude="*.lastUpdated" \
+              --exclude="resolver-status.properties" \
+              -bur build_1/io/trino build_2/io/trino > /dev/null 2>&1
+          error=$?
```

Comment:
i expect every shell script, including `ci.yml` `run` commands, to run with `set -e`, so the `error=$?` line should not be reachable in case of an error